src/pulumi_datarobot_utils/schema/custom_models.py

```diff
@@ -0,0 +1,153 @@
+    CPU_XL = ResourceBundle(name="XL", description="2 CPU | 2GB RAM", id="cpu.xlarge")
+    CPU_XXL = ResourceBundle(
+        name="XXL", description="2 CPU | 3GB RAM", id="cpu.2xlarge"
+    )
+    CPU_3XL = ResourceBundle(
+        name="3XL", description="2 CPU | 4GB RAM", id="cpu.3xlarge"
+    )
+    CPU_4XL = ResourceBundle(
+        name="4XL", description="2 CPU | 6GB RAM", id="cpu.4xlarge"
+    )
+    CPU_5XL = ResourceBundle(
+        name="5XL", description="2 CPU | 8GB RAM", id="cpu.5xlarge"
+    )
+    CPU_6XL = ResourceBundle(
+        name="6XL", description="2 CPU | 10GB RAM", id="cpu.6xlarge"
+    )
+    CPU_7XL = ResourceBundle(
+        name="7XL", description="2 CPU | 12GB RAM", id="cpu.7xlarge"
+    )
+    CPU_8XL = ResourceBundle(
+        name="8XL", description="2 CPU | 14GB RAM", id="cpu.8xlarge"
+    )
+    CPU_16XL = ResourceBundle(
+        name="16XL", description="4 CPU | 28GB RAM", id="cpu.16xlarge"
+    )
+    GPU_S = ResourceBundle(
+        name="GPU - S",
+        description="1 x NVIDIA T4 | 16GB VRAM | 4 CPU | 16GB RAM",
+        id="DRAWS_g4dn.xlarge_frac1_regular",
+    )
+    GPU_M = ResourceBundle(
+        name="GPU - M",
+        description="1 x NVIDIA T4 | 16GB VRAM | 8 CPU | 32GB RAM",
+        id="DRAWS_g4dn.2xlarge_frac1_regular",
+    )
+    GPU_L = ResourceBundle(
+        name="GPU - L",
+        description="1 x NVIDIA A10G | 24GB VRAM | 8 CPU | 32GB RAM",
+        id="DRAWS_g5.2xlarge_frac1_regular",
+    )
+    GPU_XL = ResourceBundle(
+        name="GPU - XL",
+        description="1 x NVIDIA L40S | 48GB VRAM | 4 CPU | 32GB RAM",
+        id="DRAWS_g6e.xlarge_frac1_regular",
+    )
+    GPU_XXL = ResourceBundle(
+        name="GPU - XXL",
+        description="4 x NVIDIA A10G | 96GB VRAM | 48 CPU | 192GB RAM",
+        id="DRAWS_g5.12xlarge_frac1_regular",
+    )
+    GPU_3XL = ResourceBundle(
+        name="GPU - 3XL",
+        description="4 x NVIDIA L40S | 192GB VRAM | 48 CPU | 384GB RAM",
+        id="DRAWS_g6e.12xlarge_frac1_regular",
+    )
+    GPU_4XL = ResourceBundle(
+        name="GPU - 4XL",
+        description="8 x NVIDIA A10G | 192GB VRAM | 192 CPU | 768GB RAM",
+        id="DRAWS_g5.48xlarge_frac1_regular",
+    )
+    GPU_5XL = ResourceBundle(
+        name="GPU - 5XL",
+        description="8 x NVIDIA L40S | 384GB VRAM | 192 CPU | 1.5TB RAM",
+        id="DRAWS_g6e.48xlarge_frac1_regular",
+    )
+
+
+class PredictionEnvironmentPlatforms(StrEnum):
+    AWS = "aws"
+    GCP = "gcp"
+    AZURE = "azure"
+    ON_PREMISE = "onPremise"
+    DATAROBOT = "datarobot"
+    DATAROBOT_SERVERLESS = "datarobotServerless"
+    OPEN_SHIFT = "openShift"
+    OTHER = "other"
+    SNOWFLAKE = "snowflake"
+    SAP_AI_CORE = "sapAiCore"
+
+
+class CustomModelArgs(Schema):
+    resource_name: str
+    name: str
+    replicas: int | None = None
+    description: str | None = None
+    base_environment_id: str | None = None
+    base_environment_version_id: str | None = None
```

Comment:
This comment could likely apply to more of the schemas in this PR but I wonder if there's a chance to add more class-level validations like (assuming I'm correct here) that if `base_environment_version_id` is not `None` then `base_environment_id` must _also_ be not `None`

Also very likely not something to update in this PR but maybe something as follow-up improvements?